packages/drivers/odsp-driver-definitions/src/tokenFetch.ts

```diff
@@ -41,6 +52,17 @@ export interface TokenFetchOptions {
 	 * to use to issue access token.
 	 */
 	tenantId?: string;
+
+	/**
+	 * Request that will be made using the fetched token.
+	 * - url: full request url, including query params
+	 * - method: method type
+	 * Request info may be encoded into the returned token and will be used by receiver to validate that caller is allowed to make specific call.
+	 * For example, TokenProvider implementation that returns PFT+POP token will have request method, along with hashed request path and query params
+	 * as token claims that later on can be used to validate that token is allowed to be used with corresponding request. This is additional
+	 * security feature which ensures that PFT+POP token cannot be replayed with various endpoints.
+	 */
+	readonly request?: { url: string; method: "GET" | "POST" | "PATCH" | "DELETE" | "PUT" };
```

Comment:
Are all the properties above valid (required) as well when `request` is specified?

Reply:
The only required property is "refresh". Everything else is needed for token refresh if the token is invalid for various reason. (claims and tenantId)